packages/react-sdk/src/index.tsx

```diff
@@ -172,33 +178,55 @@ export function BucketProvider({
   );
 }
 
+type RequestFeedbackOptions = Omit<
+  RequestFeedbackData,
+  "featureKey" | "featureId"
+>;
+
+type EmptyFeatureConfig = {
+  key: undefined;
+  payload: undefined;
+};
+
+type Feature<TKey extends FeatureKey> = {
+  isEnabled: boolean;
+  isLoading: boolean;
+  config:
+    | {
+        key: string;
+        payload: FeatureConfig<TKey>;
+      }
+    | EmptyFeatureConfig;
+  track: () => void;
+  requestFeedback: (opts: RequestFeedbackOptions) => void;
+};
+
 /**
  * Returns the state of a given feature for the current context, e.g.
  *
  * ```ts
  * function HuddleButton() {
- *   const {isEnabled, track} = useFeature("huddle");
+ *   const {isEnabled, config: { payload }, track} = useFeature("huddle");
  *   if (isEnabled) {
- *    return <button onClick={() => track()}>Start Huddle</button>;
- *   }
+ *    return <button onClick={() => track()}>{payload?.buttonTitle ?? "Start Huddle"}</button>;
  * }
  * ```
  */
-export function useFeature(key: FeatureKey) {
+export function useFeature<TKey extends FeatureKey>(key: TKey): Feature<TKey> {
   const {
     features: { features, isLoading },
     client,
   } = useContext<ProviderContextType>(ProviderContext);
 
   const track = () => client?.track(key);
-  const requestFeedback = (
-    opts: Omit<RequestFeedbackData, "featureKey" | "featureId">,
-  ) => client?.requestFeedback({ ...opts, featureKey: key });
+  const requestFeedback = (opts: RequestFeedbackOptions) =>
+    client?.requestFeedback({ ...opts, featureKey: key });
 
   if (isLoading) {
     return {
       isLoading,
       isEnabled: false,
+      config: {} as EmptyFeatureConfig,
```

Comment:
the "as" here shouldn't be necessary. Let's return `{key: undefined, payload: undefined}` ? or change the definition of `EmptyFeatureConfig`.

Reply:
The type definition needs the explicit `{key: undefined, payload: undefined}` in order for deconstruction to work without ts errors, but the object doesn't need actual values. I think this is fine to be honest.

Reply:
Generally, `as` is a code smell when used like this because we're not being truthful about what we're returning. I agree can seem somewhat academic in this case since {key: undefined} and {} is almost the same in the context of a destructuring operation `const {key} = {}` and `const {key} = {key: undefined}`. 

However, here's an example of using the `in` operator that could break because of the `as`:
```ts
// user expects `key` to be in the returned value if `a()` is called:
function a() {
  return {} as {key: undefined}
}

function b() {
  return {'b': true}
}


const ret: {key: string} | {'b': boolean} = myCondition ? a() : b();
.... 
// somewhere further down
if (key in ret) {
  .... // expected a was called
}
// expected b was called
```

Reply:
Is the `in` operator useful in this case? I believe we only want this construct for deconstruction purposes. I could mark the key and payload as optional as well. That should avoid the need to typecast. An object with two undefined props seems more of smell to me tbh

Reply:
My point is we're lying to our users for no real reason. They can reasonably expect the `in` operator to work here and it wont.

Maybe the definition of `EmptyFeatureConfig` can change to `{key: string, payload?: any} | {}` and then we don't need the typecast? Or maybe `{key: string | payload?: any} | Record<string, never>` because `{}` [doesn't mean the straight forward thing in Typescript](https://www.totaltypescript.com/the-empty-object-type-in-typescript).

> I could mark the key and payload as optional as well. 

also an option, but doesn't really signal that `key` is present if `payloads` is present - although that might be OK.